### ISSUE TITLE
Making perplexwrap compatible in the makefile?

### ISSUE THREAD

**Issue by sc-dyer** [jadconnolly/Perple_X#7]
I was looking at trying to get the functions of perplexwrap included in the Julia Yggdrasil build and noticed that the makefile was changed to no longer compile perplexwrap. As far as I understand, for BinaryBuilder to work, the library needs to be a target in the official repo.

I was just wondering what the primary issues were with it so that I can try to address them. Is it only the use of the fPIC flag or are there other problems?

**Comment by boriskaus** [comment]
Yes, perplexwrap needs to be rewritten in such a way that it compiles with normal compilers. Once that is done, I am happy to include it again but the way it is now it brakes the compilation process. 
Let me know when you fix it.

**Comment by sc-dyer** [comment]
That is disappointing, do you have any specific compiler flags that it needs to work with or flags that need to be removed (besides fPIC) or is the desired goal to compile it simply with -O3?

**Comment by boriskaus** [comment]
you can see the failures [here](https://buildkite.com/julialang/yggdrasil/builds/13817#01926dfb-421f-45f8-8084-b57d88b82fcb/668-18128)

```
[04:36:52] perplexwrap.f:15:72:
[04:36:52] 
[04:36:52]    15 |             subroutine initMeemum(fileName,nameLength, componentNames,sysCompo,componentMass)
[04:36:52]       |                                                                        1
[04:36:52] Error: Unexpected junk in formal argument list at (1)
[04:36:52] perplexwrap.f:22:72:
[04:36:52] 
[04:36:52]    22 |                 integer, intent(in) :: nameLength
```

This is the only file that crashes, which is why we removed it. 
If you want fix this, please update [this](https://github.com/JuliaPackaging/Yggdrasil/blob/master/P/PETSc/build_tarballs.jl) file using the [BinaryBuilder](https://github.com/JuliaPackaging/BinaryBuilder.jl) cross-compilation package.
We're happy to take it into account again once the makefile compiles all sources in clean manner (using the BB system).

**Comment by sc-dyer** [comment]
Ok that's great thank you, I will try to set up a local environment to test it with BinaryBuilder and see if I can figure this out
